src/Oracles/Dependencies.hs

```diff
@@ -81,3 +81,16 @@ dependenciesOracles = do
         putLoud $ "Reading dependencies from " ++ file ++ "..."
         contents <- map words <$> readFileLines file
         return $ Map.fromList [ (key, values) | (key:values) <- contents ]
+
+sortPkgsByDep :: [Package] -> Action [Package]
+sortPkgsByDep pkgs = do
+    elems <- mapM (\p -> (p,) <$> pkgDependencies p) pkgs
+    return $ map fst $ topSort elems
+  where
+    annotateInDeg es e =
+     (foldr (\e' s -> if fst e' `elem` snd e then s + 1 else s) 0 es, e)
+    topSort [] = []
+    topSort es =
```

Comment:
In this case, I am using it to install packages (`Rules.Install.installPackages`). But since we don't track installed artifacts and simply execute the installation repetitively, so there is no dependency information maintained.

I think, to use Shake's functionality, we need to write installPackages as a set of `Rules`, instead of `Action`.